File: src/lib/api/employees.ts

import api from "./axios";

// ✅ تایپ کامل کارمند (برای مدیریت ادمین)
export interface Employee {
  id: string;
  name: string;
  phone: string;
  email?: string;
  national_id?: string;
  role: "EMPLOYEE" | "MANAGER" | "ADMIN";
  department?: string;
  position?: string;
  avatar?: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  _count?: {
    managedEvents: number;
    tickets?: number;
  };
}

// ✅ تایپ عمومی کارمند (برای نمایش در سایت)
export interface EmployeePublic {
  id: string;
  name: string;
  department?: string;
  position?: string;
  avatar?: string;
  is_active?: boolean;  // ✅ اضافه شد
}

export const employeesAPI = {
  // 📥 دریافت لیست عمومی کارمندان (بدون احراز هویت)
  getPublic: async (): Promise<EmployeePublic[]> => {
    const response = await api.get("/employees/public");
    return response.data;
  },

  // 📥 دریافت جزئیات یک کارمند (نیاز به ادمین)
  getById: async (id: string): Promise<Employee> => {
    const token = localStorage.getItem("token") || "";
    const response = await api.get(`/employees/${id}`, {
      headers: { Authorization: `Bearer ${token}` },
    });
    return response.data;
  },

  // 📤 ایجاد کارمند جدید (نیاز به ادمین)
  create: async (data: {
    phone: string;
    national_id: string;
    department?: string;
    position?: string;
    role?: "EMPLOYEE" | "MANAGER" | "ADMIN";
  }): Promise<Employee> => {
    const token = localStorage.getItem("token") || "";
    const response = await api.post("/employees", data, {
      headers: { Authorization: `Bearer ${token}` },
    });
    return response.data;
  },

  // 📤 ویرایش کارمند (نیاز به ادمین)
  update: async (
    id: string,
    data: {
      national_id?: string;
      role?: "EMPLOYEE" | "MANAGER" | "ADMIN";
      department?: string;
      position?: string;
      is_active?: boolean;
    }
  ): Promise<Employee> => {
    const token = localStorage.getItem("token") || "";
    const response = await api.put(`/employees/${id}`, data, {
      headers: { Authorization: `Bearer ${token}` },
    });
    return response.data;
  },

  // 🗑️ حذف کارمند (نیاز به ادمین)
  delete: async (id: string): Promise<void> => {
    const token = localStorage.getItem("token") || "";
    await api.delete(`/employees/${id}`, {
      headers: { Authorization: `Bearer ${token}` },
    });
  },

  // 🔐 لاگین کارمند
  login: (data: { phone: string; password: string }) => {
    return api.post("/employees/login", data);
  },

  // 📥 دریافت لیست کامل کارمندان (نیاز به ادمین)
  getAll: async (): Promise<Employee[]> => {
    try {
      const token = localStorage.getItem("token") || "";
      const response = await api.get("/employees", {
        headers: { Authorization: `Bearer ${token}` },
      });
      return response.data;
    } catch (error) {
      console.error("❌ خطا در دریافت کارمندان:", error);
      return [];
    }
  },
};

File: app/schemas/employees.py
# app/schemas/employees.py
from pydantic import BaseModel, Field, EmailStr
from typing import Optional, List
from datetime import datetime
from enum import Enum

class EmployeeRoleEnum(str, Enum):
    EMPLOYEE = "EMPLOYEE"
    MANAGER = "MANAGER"

class EmployeeCreateSchema(BaseModel):
    phone: str = Field(..., description="شماره همراه کاربر ثبت‌نام شده", example="09129876543")
    national_id: str = Field(..., description="کد ملی کارمند", example="0012345678")
    department: Optional[str] = Field(None, description="دپارتمان یا بخش کاری", example="فنی و توسعه")
    position: Optional[str] = Field(None, description="عنوان شغلی", example="توسعه‌دهنده Senior Backend")
    role: Optional[EmployeeRoleEnum] = Field(EmployeeRoleEnum.EMPLOYEE, description="نقش سازمانی (EMPLOYEE یا MANAGER)")

class EmployeeUpdateSchema(BaseModel):
    national_id: Optional[str] = Field(None, description="کد ملی")
    role: Optional[EmployeeRoleEnum] = Field(None, description="نقش سازمانی")
    department: Optional[str] = Field(None, description="دپارتمان")
    position: Optional[str] = Field(None, description="عنوان شغلی")
    is_active: Optional[bool] = Field(None, description="وضعیت فعال/غیرفعال بودن حساب")

class EmployeePublicResponseSchema(BaseModel):
    id: str = Field(..., description="شناسه یکتا")
    name: str = Field(..., description="نام و نام خانوادگی")
    department: Optional[str] = Field(None, description="دپارتمان")
    position: Optional[str] = Field(None, description="عنوان شغلی")
    avatar: Optional[str] = Field(None, description="آدرس تصویر پروفایل")

class EmployeeDetailResponseSchema(BaseModel):
    id: str = Field(..., description="شناسه یکتای کارمند")
    name: str = Field(..., description="نام و نام خانوادگی")
    phone: str = Field(..., description="شماره همراه")
    national_id: Optional[str] = Field(None, description="کد ملی")
    email: Optional[str] = Field(None, description="ایمیل")
    role: str = Field(..., description="نقش سازمانی")
    department: Optional[str] = Field(None, description="دپارتمان")
    position: Optional[str] = Field(None, description="عنوان شغلی")
    avatar: Optional[str] = Field(None, description="آدرس تصویر پروفایل")
    is_active: bool = Field(True, description="وضعیت فعال بودن")
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None